edgelet/edgelet-http/src/unix.rs

```diff
@@ -1,45 +1,37 @@
 // Copyright (c) Microsoft. All rights reserved.
 
 use std::fs;
-#[cfg(unix)]
-use std::os::unix::fs::MetadataExt;
 use std::os::unix::prelude::PermissionsExt;
 use std::path::Path;
 
 use failure::ResultExt;
 use log::{debug, error};
 #[cfg(unix)]
-use nix::sys::stat::{umask, Mode};
-#[cfg(unix)]
-use scopeguard::defer;
-#[cfg(unix)]
 use tokio_uds::UnixListener;
 
 use crate::error::{Error, ErrorKind};
 use crate::util::{incoming::Incoming, socket_file_exists};
 
 pub fn listener<P: AsRef<Path>>(path: P, unix_socket_permission: u32) -> Result<Incoming, Error> {
     let listener = if socket_file_exists(path.as_ref()) {
```

Comment:
With this change, the arms in this conditional block are mostly identical. In one case we remove the existing file, and in the other we make sure the parent directory structure is in place, but then the rest is the same. Would it be better to consolidate than to duplicate?

```rust
pub fn listener<P: AsRef<Path>>(path: P, unix_socket_permission: u32) -> Result<Incoming, Error> {
    if socket_file_exists(path.as_ref()) {
        fs::remove_file(&path)
            .with_context(|_| ErrorKind::Path(path.as_ref().display().to_string()))?;
    } else {
        if let Some(parent) = path.as_ref().parent() {
            if !parent.exists() {
                fs::create_dir_all(parent).with_context(|err| {
                    error!("Cannot create directory, error: {}", err);
                    ErrorKind::Path(path.as_ref().display().to_string())
                })?;
            }
        }
    }

    let listener = UnixListener::bind(&path)
        .with_context(|_| ErrorKind::Path(path.as_ref().display().to_string()))?;

    fs::set_permissions(
        path.as_ref(),
        fs::Permissions::from_mode(unix_socket_permission),
    )
    .map_err(|err| {
        error!("Cannot set directory permissions: {}", err);
        ErrorKind::Path(path.as_ref().display().to_string())
    })?;

    Ok(Incoming::Unix(listener))
}
```

Reply:
It is still a draft. I want to discuss that change first. I don't understand why umask works  intermittently

Reply:
Gotcha. 
As I said to Damon it is a draft. I just wanted to discuss with you the validity of that change first.